pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -518,33 +518,6 @@ func (cd *clusterDeploy) deployAgent(cluster *apimgmtv3.Cluster) error {
 			}
 			logrus.Debugf("Ignored '%s' error during delete kube-api-auth DaemonSet", dsNotFoundError)
 		}
-		if cluster.Status.Driver != apimgmtv3.ClusterDriverRKE {
-			if output, err = kubectl.Delete([]byte(systemtemplate.NodeAgentDaemonSet), kubeConfig); err != nil {
-				logrus.Tracef("Output from kubectl delete cattle-node-agent DaemonSet, output: %s, err: %v", string(output), err)
-				// Ignore if the resource does not exist and it returns 'daemonsets.apps "kube-api-auth" not found'
-				dsNotFoundError := "daemonsets.apps \"cattle-node-agent\" not found"
-				if !strings.Contains(string(output), dsNotFoundError) {
-					return cluster, errors.WithMessage(types.NewErrors(err, errors.New(string(output))), "kubectl delete failed")
-				}
-				logrus.Debugf("Ignored '%s' error during delete cattle-node-agent DaemonSet", dsNotFoundError)
-			}
-		} else if strings.ToLower(settings.AgentRolloutWait.Get()) == "true" {
-			// Check for agent daemonset rollout if parameter is set and driverv32.ClusterDriverRKE
-			timeout := settings.AgentRolloutTimeout.Get()
-			_, err = time.ParseDuration(timeout)
-			if err != nil {
-				logrus.Warnf("[deployAgent] agent-rollout-timeout setting must be in Duration format. Using default: 300s")
-				timeout = "300s"
-			}
-			logrus.Debugf("clusterDeploy: deployAgent: waiting rollout agent daemonset for cluster [%s]", cluster.Name)
-			output, err := kubectl.RolloutStatusWithNamespace("cattle-system", "ds/cattle-node-agent", timeout, kubeConfig)
-			if err != nil {
-				logrus.Debugf("clusterDeploy: deployAgent: timeout waiting rollout agent daemonset for cluster [%s]: %v", cluster.Name, err)
-				return cluster, errors.WithMessage(types.NewErrors(err, errors.New(formatKubectlApplyOutput(string(output)))), "Timeout waiting rollout agent daemonset")
-			}
-			logrus.Debugf("clusterDeploy: deployAgent: successfully rollout agent daemonset for cluster [%s]", cluster.Name)
-			apimgmtv3.ClusterConditionAgentDeployed.Message(cluster, "Successfully rollout agent daemonset")
-		}
 		apimgmtv3.ClusterConditionAgentDeployed.Message(cluster, string(output))
 		return cluster, nil
 	}); err != nil {
```

Comment:
We can also remove the `getNodeAgentImage` function, and no longer cache it in this handler.